unitTests/ui/DynamicFacet/DynamicFacetTest.ts

```diff
@@ -167,20 +171,52 @@ export function DynamicFacetTest() {
       expect(test.cmp.queryController.executeQuery).toHaveBeenCalled();
     });
 
+    it('allows to trigger a new isolated query', () => {
+      spyOn(test.cmp.dynamicFacetQueryController, 'executeIsolatedQuery');
+      const beforeExecuteQuery = jasmine.createSpy('beforeExecuteQuery', () => {});
+      test.cmp.ensureDom();
+      test.cmp.triggerNewIsolatedQuery(beforeExecuteQuery);
+
+      expect(test.cmp.dynamicFacetQueryController.executeIsolatedQuery).toHaveBeenCalled();
+      expect(beforeExecuteQuery).toHaveBeenCalled();
+      expect(test.cmp.values.render).toHaveBeenCalled();
+    });
+
+    it('triggering a new isolated query updates the values', () => {
+      test.cmp.ensureDom();
+      test.cmp.triggerNewIsolatedQuery();
+
+      expect(test.cmp.values.render).toHaveBeenCalled();
+    });
+
     it('allows to reset', () => {
       mockFacetValues[1].state = FacetValueState.selected;
       mockFacetValues[3].state = FacetValueState.selected;
       initializeComponent();
-      test.cmp.ensureDom();
-      expect(test.cmp.values.selectedValues.length).toBe(2);
-
       test.cmp.reset();
 
       expect($$(test.cmp.element).hasClass('coveo-active')).toBe(false);
       expect(test.cmp.values.selectedValues.length).toBe(0);
       testQueryStateModelValues();
     });
 
+    it('when calling reset, should clear and rerender values if there is any active', () => {
+      mockFacetValues[0].state = FacetValueState.selected;
+      initializeComponent();
+
+      test.cmp.reset();
+
+      expect(test.cmp.values.clearAll).toHaveBeenCalled();
+      expect(test.cmp.values.render).toHaveBeenCalledTimes(2);
+    });
+
+    it('when calling reset, should not clear and rerender values if there is any active', () => {
```

Comment:
Seems that this test description should say `...when there are no active values`?